firmware/controllers/gauges/tachometer.cpp

```diff
@@ -18,26 +18,71 @@
 EXTERN_ENGINE;
 
 static scheduling_s tachTurnSignalOff;
+static float halfperiodMs = 4; /* period in ms, used for multi pulse output, actually half period */
+static uint8_t isStarted = 0;
+
+static void toggleTachPinHigh(void);
+static void toggleTachPinLow(void);
 
 static void turnTachPinLow(void) {
 	enginePins.tachOut.setLow();
 }
 
+static void toggleTachPinLow(void)
+{
+	if(isStarted)
+	{
+		enginePins.tachOut.setLow();
+		engine->executor.scheduleForLater(&tachTurnSignalOff, (int)MS2US(halfperiodMs), (schfunc_t) &toggleTachPinHigh, NULL);
```

Comment:
there's a race condition here - I'd love to discuss what the correct way to implement this would be.  Can you join our Slack so we can discuss better there?  https://rusefi.com/forum/viewtopic.php?t=1198